packages/react/src/components/Authenticator/shared/SignInSignUpTabs.tsx

```diff
@@ -32,7 +32,11 @@ export const SignInSignUpTabs = ({
         <Tabs.Container
           value={route}
           isLazy
-          onValueChange={() => (route === 'signIn' ? toSignUp() : toSignIn())}
+          onValueChange={(prevRoute) => {
+            if (prevRoute !== route) {
+              route === 'signIn' ? toSignUp() : toSignIn();
```

Comment:
I don't know why, but this is hurting my brain. Is the route the tab I clicked on or the route I'm currently at? So if I click any tab then it will essentially toggle to the other one? Why is comparing the prevRoute to the current route necessary?

Reply:
I think these variable names are switched. It looks like `route` is the active tab and `prevRoute` is the clicked tab.

So the old logic was just "if Sign In is the active tab go to Sign Up, otherwise go to Sign In", which results in always going to the other tab no matter what is clicked.

And the new logic is just adding the check of "if the clicked tab isn't the active tab...". It's still just flipping the value without regard for what was clicked, but as long as there are only the two tabs it will continue to work.

Reply:
`newRoute` might be the more consistent name, since TabContainer's `setActiveTab()` is using the name `newValue` for the argument passed to `onValueChange()`

Reply:
Updated in latest to use newRoute instead, that makes more sense for the actual logic happening.